src/test/util/time.h

```diff
@@ -5,19 +5,57 @@
 #ifndef BITCOIN_TEST_UTIL_TIME_H
 #define BITCOIN_TEST_UTIL_TIME_H
 
+#include <util/check.h>
 #include <util/time.h>
 
-struct ElapseSteady {
+#include <optional>
+
+class ElapseSteady
+{
     MockableSteadyClock::mock_time_point::duration t{MockableSteadyClock::INITIAL_MOCK_TIME};
-    ElapseSteady()
-    {
-        (*this)(0s); // init
-    }
+
+public:
+    /** Initialize with INITIAL_MOCK_TIME. */
+    ElapseSteady() { (*this)(0s); }
```

Comment:
What would happen if we ran multiple such tests on different threads? Could we guard against that via something like:
```suggestion
    ElapseSteady()
    {
        Assert(GetMockTime() == 0s); // Ensure that the steady clock is not mocked yet.
        MockableSteadyClock::SetMockTime(t);
    }
```
nit: `(*this)(0s);` is not very intuitive in my opinion